app/Http/Controllers/api/BalanceEntryController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Http\Controllers\api;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+
+use App\BalanceEntry;
+
+use Illuminate\Support\Facades\DB;
+
+class BalanceEntryController extends Controller
+{
+    public function index(Request $request)
+    {
+        $result = BalanceEntry::all()->sortByDesc('date')->values()->all();
+        if ($request->input('grouped_by_date')) {
+            $entries = [];
+            foreach ($result as $row) {
+                $date = date("n/j/Y", strtotime($row->date));
+                $entries[$date][] = $row;
+            }
+
+            $result = $entries;
```

Comment:
This could probably be cleaned up by using the collection `groupBy` method, or even grouping using a query instead of the collection.